language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConfigurableCodeAction.java

```diff
@@ -0,0 +1,346 @@
+                                               String confName, Position positionToAddConfVar,
+                                               boolean addNewLineAtStart) {
+        String typeDescriptor = FunctionGenerator.getReturnTypeAsString(context, typeSymbol.signature());
+        String value = node.toSourceCode().strip();
+        LineRange replaceRange = node.lineRange();
+        String confDeclaration = addNewLineAtStart ? System.lineSeparator() : "";
+        confDeclaration += String.format("configurable %s %s = %s;%n", typeDescriptor, confName, value);
+
+        TextEdit confDecEdit = new TextEdit(new Range(positionToAddConfVar, positionToAddConfVar), confDeclaration);
+        TextEdit replaceEdit = new TextEdit(new Range(PositionUtil.toPosition(replaceRange.startLine()),
+                PositionUtil.toPosition(replaceRange.endLine())), confName);
+        return List.of(confDecEdit, replaceEdit);
+    }
+
+    private static ConfigurableData getConfigurableData(CodeActionContext context) {
+        ModulePartNode modulePartNode = context.currentSyntaxTree().get().rootNode();
+        NodeList<ImportDeclarationNode> importsList = modulePartNode.imports();
+
+        ModuleVariableDeclarationNode configurableVarNode = null;
+        for (Node node: modulePartNode.children()) {
+            if (node.kind() == SyntaxKind.MODULE_VAR_DECL) {
+                ModuleVariableDeclarationNode modVarDeclarationNode = (ModuleVariableDeclarationNode) node;
+                if (hasConfigurableQualifier(modVarDeclarationNode)) {
+                    configurableVarNode = modVarDeclarationNode;
+                    break;
+                }
+            }
+        }
+
+        if (configurableVarNode != null) {
+            return new ConfigurableData(new Position(configurableVarNode.lineRange().startLine().line(), 0),
+                    !configurableVarNode.toString().startsWith(System.lineSeparator()));
+        } else if (importsList.isEmpty()) {
+            return new ConfigurableData(
+                    PositionUtil.toPosition(modulePartNode.lineRange().startLine()), false);
+        }
+        ImportDeclarationNode lastImport = importsList.get(importsList.size() - 1);
+        return new ConfigurableData(
+                new Position(lastImport.lineRange().endLine().line() + 1, 0), true);
+    }
+
+    private static String getConfigurableName(CodeActionContext context) {
+        Position pos = context.range().getEnd();
+        Set<String> allNames = context.visibleSymbols(new Position(pos.getLine(), pos.getCharacter())).stream()
+                .map(Symbol::getName)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet());
+
+        return NameUtil.generateTypeName(CONFIGURABLE_NAME_PREFIX, allNames);
+    }
+
+    private static String getConfigurableNameUsingParamName(CodeActionContext context, String paramName) {
```

Comment:
We can use the same name since `method overloading`
```suggestion
    private static String getConfigurableName(CodeActionContext context, String paramName) {
```